framework/aster-frame/src/trap/irq.rs

```diff
@@ -118,41 +118,62 @@ pub fn disable_local() -> DisabledLocalIrqGuard {
 
 /// A guard for disabled local IRQs.
 pub struct DisabledLocalIrqGuard {
-    was_enabled: bool,
+    /// Whether the local IRQs need to be re-enabled when this gaurd is dropped.
+    /// Usually depends on whether the local IRQs were enabled at the moment the guard was created.
+    need_reenable: bool,
     preempt_guard: DisablePreemptGuard,
 }
 
 impl !Send for DisabledLocalIrqGuard {}
 
+/// Disable local IRQs if they are enabled.
+/// Return whether the local IRQs were enabled before this function call.
+pub fn disable_irq_if_enabled() -> bool {
+    let was_enabled = irq::is_local_enabled();
+    if was_enabled {
+        irq::disable_local();
+    }
+    was_enabled
+}
+
+/// Enable local IRQs if they are disabled.
+/// Return whether the local IRQs were enabled before this function call.
+pub fn enable_irq_if_disabled() -> bool {
+    let was_enabled = irq::is_local_enabled();
+    if !was_enabled {
+        irq::enable_local();
+    }
+    was_enabled
+}
+
 impl DisabledLocalIrqGuard {
     fn new() -> Self {
-        let was_enabled = irq::is_local_enabled();
-        if was_enabled {
-            irq::disable_local();
-        }
-        let preempt_guard = disable_preempt();
         Self {
-            was_enabled,
-            preempt_guard,
+            need_reenable: disable_irq_if_enabled(),
+            preempt_guard: disable_preempt(),
         }
     }
 
     /// Transfer the saved IRQ status of this guard to a new guard.
     /// The saved IRQ status of this guard is cleared.
+    /// A gaurd after this function call will not re-enable IRQs when dropped.
     pub fn transfer_to(&mut self) -> Self {
-        let was_enabled = self.was_enabled;
-        self.was_enabled = false;
+        let need_reenable = self.need_reenable;
+        self.need_reenable = false; // avoid enabling IRQs in `drop`
         Self {
-            was_enabled,
+            // The right to re-enable irqs on `drop` is yielded to the caller
+            need_reenable,
             preempt_guard: disable_preempt(),
         }
     }
 }
 
 impl Drop for DisabledLocalIrqGuard {
     fn drop(&mut self) {
-        if self.was_enabled {
-            irq::enable_local();
+        if self.need_reenable {
+            // Irqs will not be re-enabled if irqs were disabled
+            // when the gaurd is created.
```

Comment:
```suggestion
            // IRQs will not be re-enabled if IRQs were disabled
            // when the gaurd is created.
```
I suggest keeping the words in comments consistent and in plain English.